br/pkg/lightning/mydump/parquet_parser_test.go

```diff
@@ -327,3 +336,40 @@ func TestNsecOutSideRange(t *testing.T) {
 	// carry operation. i.e. 1000000000 nsec => 1 sec
 	require.Equal(t, a.Add(1*time.Second), b)
 }
+
+func TestOpenParquetReaderWithThreshold(t *testing.T) {
+	dir := t.TempDir()
+	name := "test123.parquet"
+	// prepare data
+	prepareSimpleData(t, dir, name)
+
+	store, err := storage.NewLocalStorage(dir)
+	require.NoError(t, err)
+
+	testPath := filepath.Join(dir, name)
+	fileInfo, err := os.Stat(testPath)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// doesn't preload all the data
+	pqtFile1, err := OpenParquetReader(ctx, store, name, fileInfo.Size(), 0)
+	require.NoError(t, err)
+
+	parser1, err := NewParquetParser(ctx, store, pqtFile1, name)
+	require.NoError(t, err)
+	defer parser1.Close()
+
+	verifySimpleData(t, parser1)
```

Comment:
nit: should we check the actual structure here to make sure the logic works out?